addon/components/one-way-checkbox.js

```diff
@@ -15,29 +15,41 @@ const OneWayCheckboxComponent = Component.extend(DynamicAttributeBindings, {
   NON_ATTRIBUTE_BOUND_PROPS: ['update'],
 
   attributeBindings: [
-    'isChecked:checked',
     'type',
     'value'
   ],
 
+  checkedValue: computed('paramChecked', 'checked', function() {
+    let checkedValue = get(this, 'paramChecked');
+
+    if (checkedValue === undefined) {
+      return checkedValue = get(this, 'checked');
+    }
+
+    return checkedValue;
+  }),
+
   didInsertElement() {
     this._super(...arguments);
     this.element.addEventListener('click', (e) => this._click(e));
-  },
 
-  didReceiveAttrs() {
-    this._super(...arguments);
-
-    let value = get(this, 'paramChecked');
-    if (value === undefined) {
-      value = get(this, 'checked');
+    const checkedValue = get(this, 'checkedValue');
+    if (checkedValue) {
+      this.element.setAttribute('checked', 'checked');
     }
+  },
 
-    set(this, 'isChecked', value);
+  didUpdateAttrs() {
+    let checkedValue = get(this, 'checkedValue');
+    this.element.checked = checkedValue;
   },
 
   _click(event) {
-    invokeAction(this, 'update', this.readDOMAttr('checked'), event);
+    let checkedProp = this.element.checked;
+    invokeAction(this, 'update', checkedProp, event);
+
+    let checkedValue = get(this, 'checkedValue');
+    this.element.checked = checkedValue;
```

Comment:
Update with the `checked` _property_ instead of the _attribute_, then update the `_property_` based on the value after `update()` has called